packages/x-charts/src/ChartsOnClickHandler/ChartsOnClickHandler.tsx

```diff
@@ -0,0 +1,116 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { SVGContext } from '../context/DrawingProvider';
+import { InteractionContext, ItemInteractionData } from '../context/InteractionProvider';
+import { CartesianContext } from '../context/CartesianContextProvider';
+import { FormattedSeries, SeriesContext } from '../context/SeriesContextProvider';
+import { ChartSeriesDefaultized, ChartSeriesType } from '../models/seriesType/config';
+
+type ItemData = {
+  dataIndex?: number;
+  series: ChartSeriesDefaultized<ChartSeriesType>;
+};
+
+type AxisData = {
+  dataIndex: number;
+  axisValue?: number | Date | string;
+  seriesValues: Record<string, number | null | undefined>;
+};
+
+export type OnClickData = {
+  item: null | ItemData;
+  axis: null | AxisData;
+};
+
+export interface ChartsOnClickHandlerProps {
+  /**
+   * The function called for onCLick events.
+   * @param {MouseEvent} event The mouse event catched by `<svg/>`.
+   * @param {OnClickData} data The data linked to click with shape `{item, axis}`.
+   */
+  onClick?: (event: MouseEvent, data: OnClickData) => void;
+}
+
+function getItemData(
+  item: null | ItemInteractionData<ChartSeriesType>,
+  series: FormattedSeries,
+): null | {
+  dataIndex?: number;
+  series: ChartSeriesDefaultized<ChartSeriesType>;
+} {
+  if (item === null) {
+    return null;
+  }
+  return { series: series[item.type]?.series[item.seriesId]!, dataIndex: item.dataIndex };
```

Comment:
Are we sure about this `!`?
If we are not confident enough and we use `?` before, then the value can probably be undefined.